src/colorizer/Track.ts

```diff
@@ -2,21 +2,26 @@ export default class Track {
   public trackId: number;
   public times: number[];
   public ids: number[];
-  // TODO:  centroids
+  // Centroids must be packed as 2D array for sorting to work correctly in
+  // the constructor.
+  public centroids: number[][] | undefined;
 
-  constructor(trackId: number, times: number[], ids: number[]) {
+  constructor(trackId: number, times: number[], ids: number[], centroids?: number[][]) {
     this.trackId = trackId;
     this.times = times;
     this.ids = ids;
+    this.centroids = centroids;
 
-    // sort both, ascending by time
-    // I have no idea if the time domain would be presorted or not
+    // sort time, id, and centroids, ascending by time
     const shouldSort = true;
     if (shouldSort) {
       const indices = [...times.keys()];
       indices.sort((a, b) => (times[a] < times[b] ? -1 : times[a] === times[b] ? 0 : 1));
       this.times = indices.map((i) => times[i]);
       this.ids = indices.map((i) => ids[i]);
+      if (centroids) {
+        this.centroids = indices.map((i) => centroids[i]);
+      }
```

Comment:
here is how you do it while keeping a 1D array:
```
this.centroids = indices.reduce(function(result, i) {
  result.push(centroids[i*2], centroids[i*2+1]);
  return result;
}, []);

```

Reply:
May also possible to use `flatMap` here?

Reply:
omg yes.  flatMap is WAY better isn't it.  Thank you.